mono/mini/mini-runtime.c

```diff
@@ -2008,15 +2017,25 @@ wait_or_register_method_to_compile (MonoMethod *method, MonoDomain *domain)
 			++entry->threads_waiting;
 
 			g_assert (entry->has_cond);
-			mono_coop_cond_wait (&entry->cond, &compilation_data.lock);
+			mono_coop_cond_timedwait (&entry->cond, &compilation_data.lock, MAX_JIT_TIMEOUT_MS);
 			--entry->threads_waiting;
```

Comment:
Doesn't this going to mask bugs ?

Reply:
Yes, it will.
OTH, it ensures progress can be made even if slow enough. Do you think it's not worth it?